src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -564,6 +564,15 @@ module Crystal
             if default_value = member.default_value
               counter = interpret_enum_value(default_value, enum_base_type)
             end
+
+            if default_value.is_a?(Crystal::NumberLiteral)
+              enum_base_kind = enum_base_type.kind
+              node_kind = return_enum_kind(default_value)
+              if (enum_base_kind == :i32) && (enum_base_kind != node_kind)
```

Comment:
maybe was my mistake... now I run the specs and it's passing, let me rebuild the compiler and run all spec to make sure build won't break, I'm pushing in a few minutes 👍